modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverter.java

```diff
@@ -824,60 +842,142 @@ private Type convertIndexType(IgniteSqlIndexType type) {
                     // TODO: IGNITE-17376
                     throw new UnsupportedOperationException("Type is not supported: " + columnType);
-                case INT32:
-                    return literal.getValueAs(Integer.class);
-                case INT64:
-                    return literal.getValueAs(Long.class);
-                case INT16:
-                    return literal.getValueAs(Short.class);
-                case INT8:
-                    return literal.getValueAs(Byte.class);
+                case INT32: {
+                    acceptStringOrNumericLiteral(literal, columnType);
+                    long val = literal.longValue(false);
+                    return convertToIntExact(val);
+                }
+                case INT64: {
+                    acceptStringOrNumericLiteral(literal, columnType);
+                    BigDecimal val = literal.bigDecimalValue();
+                    return convertToLongExact(Objects.requireNonNull(val));
+                }
+                case INT16: {
+                    acceptStringOrNumericLiteral(literal, columnType);
+                    long val = literal.longValue(false);
+                    return convertToShortExact(val);
+                }
+                case INT8: {
+                    acceptStringOrNumericLiteral(literal, columnType);
+                    long val = literal.longValue(false);
+                    return convertToByteExact(val);
+                }
                 case DECIMAL:
+                    acceptStringOrNumericLiteral(literal, columnType);
                     return literal.getValueAs(BigDecimal.class);
                 case DOUBLE:
+                    acceptStringOrNumericLiteral(literal, columnType);
                     return literal.getValueAs(Double.class);
                 case FLOAT:
+                    acceptStringOrNumericLiteral(literal, columnType);
                     return literal.getValueAs(Float.class);
                 case BYTE_ARRAY:
-                    return literal.getValueAs(byte[].class);
+                    byte[] arr = literal.getValueAs(byte[].class);
+                    // varbinary without limitation
+                    if (precision != PRECISION_NOT_SPECIFIED && Objects.requireNonNull(arr).length > precision) {
+                        throw new SqlException(STMT_VALIDATION_ERR,
+                                format("Value too long for type binary({})", precision));
+                    }
+                    return arr;
                 case BOOLEAN:
-                    return literal.getValueAs(Boolean.class);
+                    String val = Objects.requireNonNull(literal.toValue());
+                    val = val.trim().toLowerCase();
+
+                    if ("true".equals(val) || "yes".equals(val) || "1".equals(val)) {
+                        return true;
+                    }
+
+                    if ("false".equals(val) || "no".equals(val) || "0".equals(val)) {
```

Comment:
`numeric` to `boolean` is not a legal type transformation - both standard disallows iand current implementation reject it.

> sql("SELECT 1::BOOLEAN");
>Err: Cast function cannot convert value of type INTEGER to type BOOLEAN

Reply:
`no` `yes` are also not considered legal.

Reply:
ok i will check standard but .. i think that cast differs from default expressions and also this how it works in PG.

Reply:
it's Ignite, not PG

Reply:
fixed